packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -0,0 +1,270 @@
+import iam = require('@aws-cdk/aws-iam');
+import kms = require('@aws-cdk/aws-kms');
+import cdk = require('@aws-cdk/cdk');
+import { SecretString } from './secret-string';
+import secretsmanager = require('./secretsmanager.generated');
+
+/**
+ * A secret in AWS Secrets Manager.
+ */
+export interface ISecret extends cdk.IConstruct {
+  /**
+   * The customer-managed encryption key that is used to encrypt this secret, if any. When not specified, the default
+   * KMS key for the account and region is being used.
+   */
+  readonly encryptionKey?: kms.IEncryptionKey;
+
+  /**
+   * The ARN of the secret in AWS Secrets Manager.
+   */
+  readonly secretArn: string;
+
+  /**
+   * Returns a SecretString corresponding to this secret, so that the secret value can be referred to from other parts
+   * of the application (such as an RDS instance's master user password property).
+   */
+  asSecretString(): SecretString;
```

Comment:
rename to `toSecretString`. We use the "asXXX" notation for inversion of control methods and users are not supposed to use them directly.